pkg/miniogw/gateway-storj.go

```diff
@@ -38,17 +152,49 @@ func (s *Storj) Name() string {
 
 // NewGatewayLayer implements cmd.Gateway
 func (s *Storj) NewGatewayLayer(creds auth.Credentials) (
-	cmd.ObjectLayer, error) {
-	return &storjObjects{}, nil
+	minio.ObjectLayer, error) {
+	return &storjObjects{storj: s}, nil
 }
 
 // Production implements cmd.Gateway
 func (s *Storj) Production() bool {
 	return false
 }
 
+//createSampleBucketList function initializes sample buckets and files in each bucket
+func (s *Storj) createSampleBucketList() {
+	s.bucketlist = s.bucketlist[:0]
```

Comment:
What does this line do?

Reply:
it initializes the slice.  Is that a comment or a clarification ?

Reply:
IMHO, this is a pretty confusing way for initializing an empty slice. Since you know the expected size of the slice (10), it would be much better to initialize it like `s.bucketlist = make([]S3BucketList, 10)`. This will avoid resizing the internal slice buffer when adding items. Even better, later in the for-loop you can get rid of the `append` and set the items directly at the designated index, i.e. `s.bucketlist[i] = S3BucketList{...}`.

Reply:
wilco

Reply:
simplified the function (rewritten)